Marsiple/Common/Network/DataFetcher.swift

```diff
@@ -0,0 +1,52 @@
+//
+//  DataFetcher.swift
+//  Marsiple
+//
+//  Created by Matej Korman on 23/07/2018.
+//  Copyright © 2018 Matej Korman. All rights reserved.
+//
+
+import Foundation
+
+class DataFetcher {
+    static func getPosts(success: @escaping ([Post]) -> Void, failure: @escaping (LocalizedError) -> Void) {
+        guard let url = MartianApi.url?.appendingPathComponent(MartianApi.URLs.posts)
+            else { return DispatchQueue.main.async { failure(DataFetcherError.urlCreationFailure) } }
+        
+        getData(url: url, method: HTTPRequestMethod.get, success: success, failure: failure)
+    }
+    
+    static func getComments(forPostId id: Int, success: @escaping ([Comment]) -> Void, failure: @escaping (LocalizedError) -> Void) {
+        guard let url = MartianApi.url?
+                            .appendingPathComponent(MartianApi.URLs.posts)
+                            .appendingPathComponent("\(id)")
+                            .appendingPathComponent(MartianApi.URLs.comments)
+            else { return DispatchQueue.main.async { failure(DataFetcherError.urlCreationFailure) } }
+        getData(url: url, method: HTTPRequestMethod.get, success: success, failure: failure)
+    }
+}
+
+private extension DataFetcher {
+    static func getData<DataType: Codable>(url: URL,
+                                           method: String,
+                                           success: @escaping ([DataType]) -> Void,
+                                           failure: @escaping (LocalizedError) -> Void) {
+        var request = URLRequest(url: url)
+        request.httpMethod = method
+        request.addValue(MartianApi.Headers.contentTypeValue, forHTTPHeaderField: HTTPRequestHeader.contentType)
```

Comment:
Just noticed this one - I think we could name this one better than it currently is. Basically, what we're doing here, is we're extracting the `default` value for our `Content-Type` header. 

Thus, we could use `MartianApi.Headers.contentTypeValue` as `MartianApi.Headers.defaultContentType` or similiar. What do you think?

Reply:
Ok. Looks more descriptive.